frontend/src/APIRequestFunction.tsx

```diff
@@ -0,0 +1,23 @@
+import store from './store';
+
+export const SERVER = window.location.hostname === 'localhost' ? 'http://localhost:3000' : '';
```

Comment:
Ich glaube unser Server läuft auf 5000 ... sieht aus wie aus Ethic-Games oder GradeMe copy-pasted ;)

Frag mal bitte Benito ob ihm die Lösung sauber genug ist. Er müsste sich mit sowas auskennen .. ich würde denken das was ich da mal implementiert hatte ist unsauber :D

Reply:
Ja, wir laufen auf 5000. Ist zwar nicht sauber, aber ich würde das an der Stelle hardcoded lassen. Sauberer wäre, wenn Server und client das aus der gleichen config auslesen würden. Da hat aber echt niemand jz Lust drauf. Aber bitte window.location.origin statt '' nehmen, da ist das Protokoll, der Hostname und der Port bei. Finde ich besser, wenn im Server tatsächlich immer der Server steht.

Reply:
ja tatsächlich von unseren frühreren projekten abgeguckt xD

Habs jetzt nach de wünschen angepasst

Reply:
Hab grad gesehen, dass der Server automatisch immer vorne an den reqPfad angehangen wird  durch die fetchfunktion.
Heißt wenn ich da zusätzlich noch den Server ranhänge sieht die anfrage ungefährt so aus: "http://localhost:5000http://localhost:5000/api/facharbeit/test"

Reply:
Wenn du keinen absoluten Pfad angibst dann ja. 
Aber wenn du lokal testen willst, dann läuft ja der Server auf port 5000 und der client auf ich glaube port 3000 ... jedenfalls auf nem anderen Port.
um lokales development zu ermöglichen muss eben wenn das ding auf localhost läuft der posrt vom server angesprochen werden und nicht der port von dem fake-server der den client bereitstellt
hoffe das ist verständlich. 
also in prod reicht es einfach den relativen pfad reinzuyeeten da der client von unserem server bereitgestellt wird
und in dev muss vorher eben noch Localhost:5000 stehen, da er die anfrage sonst an localhost:3000 schicken würde.

Reply:
man kann in der package.json einen proxy angeben. Dann schreibt man in die Request nur noch z.B. "/api/students". Vielleicht hilft euch das?
Siehe hier: https://create-react-app.dev/docs/proxying-api-requests-in-development/

Reply:
hab das jetzt mit dem path.join gemacht. denke das ist eine akzeptable lösung